context/RoutingContext.js

```diff
@@ -0,0 +1,88 @@
+import { createContext, useState, useEffect } from "react";
+import { useRouter } from "next/router";
+
+export const RoutingContext = createContext();
+
+export const RoutingProvider = ({ children }) => {
+  const [currentAsPath, setCurrentAsPath] = useState();
+  const [currentRoute, setCurrentRoute] = useState();
+  const [prevAsPath, setPrevAsPath] = useState();
+  const [queryString, setQueryString] = useState("");
+
+  const { asPath, route, query, pathname } = useRouter();
+
+  const [tagId, setTagId] = useState();
+
+  // We need asPath to have all details, but only change when route changes
+  useEffect(() => {
+    if (route !== currentRoute) {
+      setPrevAsPath(currentAsPath);
+      setCurrentAsPath(asPath);
+      setCurrentRoute(route);
+    }
+  }, [route, currentRoute, asPath]);
+
+  const setSearch = (search) => {
+    /*
```

Comment:
Ah, yes, I wanted to discuss this, but didn't investigate options yet, I agree these should defintely be adressed prior merging.
It became clear that the current approach with the tracking decorators wouldn't really be suitable for further development, and also was not really comfortable to work with.

During removal of the custom babel configuration I had not decided yet what to do with them, but did not want to remove all the references to tracking functionality upfront, in case there might be a simple way to adapt them.

Im open for suggestions here - I could just remove them from now, as we could also lookup references from the existing branches - Also, lot's of them seem obsolete and could probably better be abstracted in the hook for the respective action instead.

In the long run, I would need to research how to use react-tracking with hooks now, and maybe also evauate if it still is a good choice.
Would tracking be a priority for an initial merge?

Reply:
Regarding the commented component-library imports, they where from a point when there was no decision regarding removal and I didn't know yet if I could actually just reimplement in here, or would need to push those changes to the libary instead. Will remove them.

Reply:
I will need to recheck the purpose of `setSearch`, I don't recall if removing is actually okay or if I might need to rewrite that instead.

Reply:
sounds good 👍